src/Carnac.Logic/MessageController.cs

```diff
@@ -26,9 +27,15 @@ public KeysController(ObservableCollection<Message> keys, IMessageProvider messa
 
         public void Start()
         {
-            var messageStream = messageProvider.GetMessageStream(keyProvider.GetKeyStream()).Publish().RefCount();
+            if (stopCarnacDisposable != null) throw new InvalidOperationException("Carnac cannot be started more than once");
```

Comment:
I'm a much bigger fan of BooleanDisposable for this.  It's "nicer" to check the IsDisposed than a null check.  Minor though